### PR TITLE
Speculativly test :has-slotted()

### PR DESCRIPTION
**Pull request by Westbrook**
As per https://github.com/w3c/csswg-drafts/pull/10586 `:has-slotted()` is coming!

This adds parsing tests as well as some red/green tests for actual delivery of the selectors in practice.

### CODE REVIEW

**Comment by Westbrook** [comment]
cc: @michaelwarren1106 @JRJurman @thescientist13 @keithamus

**Review by keithamus — APPROVED** [review]
These look like good tests to get started with an implementation to me, so it gets my approval, but I'll let one of the spec editors decide if there's anything more we need to do here before merging.

**Comment by Westbrook** [comment]
@tabatkins I've got a couple of additional tests for this feature after watching @keithamus work through some WebKit implementation on this feature, would it be better to add them to this PR while it's open or to get this merged in and open a new one?

**Comment by Westbrook** [comment]
@tabatkins let me know if there is anything else I can do here, or in related issues, to support moving this forward. Web Components developers the community over really look forward to a future where these tests are green x-browser.

**Comment by Westbrook** [comment]
If anyone was interested, I have some follow up tests that build on this in https://github.com/Westbrook/wpt/pull/1. I've not added them here in hopes of getting a rough "you're not breaking anything" on these tests first. If it's preferred, I can merge the related PR and the changes will flow right into this one, further proving out the `:has-slotted` and `:has-slotted(...)` APIs. 🤞🏼

**Review by keithamus — APPROVED** [review]
Great 👍

File: css/css-scoping/has-slotted-001.tentative.html
Comment (keithamus) [inline css/css-scoping/has-slotted-001.tentative.html:8]: I think these need to be re-indented with tabs to pass lint.

Comment (Westbrook) [reply css/css-scoping/has-slotted-001.tentative.html:8]: 🙈 Every time!

**Pull request merged**
